File: tests/org/elixir_lang/reference/callable/Issue500Test.java
package org.elixir_lang.reference.callable;

import com.intellij.psi.PsiElement;
import com.intellij.testFramework.fixtures.LightCodeInsightFixtureTestCase;
import org.elixir_lang.psi.call.Call;

import static org.elixir_lang.reference.Callable.isVariable;

public class Issue500Test extends LightCodeInsightFixtureTestCase {
    /*
     * Tests
     */

    public void testIsVariable() {
        myFixture.configureByFiles("is_variable.ex");
        @SuppressWarnings("ConstantConditions") PsiElement callable = myFixture
                .getFile()
                .findElementAt(myFixture.getCaretOffset())
                .getParent()
                .getParent();

        assertInstanceOf(callable, Call.class);
        assertFalse("parameter in tuple after Alias dot is incorrectly marked as a variable", isVariable(callable));
    }

    /*
     * Protected Instance Methods
     */

    @Override
    protected String getTestDataPath() {
        return "testData/org/elixir_lang/reference/callable/issue_500";
    }
}

File: tests/org/elixir_lang/reference/callable/Issue500Test.kt
package org.elixir_lang.reference.callable

import org.elixir_lang.PlatformTestCase
import org.elixir_lang.psi.call.Call
import org.elixir_lang.reference.Callable.Companion.isVariable

class Issue500Test : PlatformTestCase() {
    fun testIsVariable() {
        myFixture.configureByFiles("is_variable.ex")
        val callable = myFixture
                .file
                .findElementAt(myFixture.caretOffset)!!
                .parent
                .parent
        assertInstanceOf(callable, Call::class.java)
        assertFalse("parameter in tuple after Alias dot is incorrectly marked as a variable", isVariable(callable))
    }

    override fun getTestDataPath(): String = "testData/org/elixir_lang/reference/callable/issue_500"
}
